### PR TITLE
feat: add map packages

### PR DESCRIPTION
**Pull request by taikitanaka3**
Related Issue
#1

Related Link
This PR depends on #4

Description
Add map packages from https://github.com/tier4/AutowareArchitectureProposal.iv

How to review
Check the following items

lanelet2_extension
map_loader
map_tf_generator

### CODE REVIEW

File: map/lanelet2_extension/README.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+Code for this explains how this lanelet2_extension library is used.
+The executable is not meant to do anything.
+
+### autoware_lanelet2_extension
+
+This node checks if an .osm file follows the Autoware version of Lanelet2 format.
+You can check by running:
+
+```sh
+rosrun lanelet2_extension autoware_lanelet2_validation _map_file:=<path/to/map.osm>
```
Comment (tkimura4) [inline map/lanelet2_extension/README.md:86]: rosrun -> ros2 run

File: map/lanelet2_extension/README.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+#### Message Conversion
+
+This contains functions to convert lanelet map objects into ROS messages.
+Currently it contains following conversions:
+
+- lanelet::LaneletMapPtr to/from lanelet_msgs::MapBinMsg
+- lanelet::Point3d to geometry_msgs::Point
+- lanelet::Point2d to geometry_msgs::Point
+- lanelet::BasicPoint3d to geometry_msgs::Point
```
Comment (tkimura4) [inline map/lanelet2_extension/README.md:52]: This description is old.

File: map/lanelet2_extension/README.md (hunk 1 above)
Comment (taikitanaka3) [reply map/lanelet2_extension/README.md]: 87cba84

File: map/lanelet2_extension/README.md (hunk 2 above)
Comment (taikitanaka3) [reply map/lanelet2_extension/README.md]: 87cba84
lanelet_msgs::MapBinMsg -> autoware_auto_mapping_msgs::msg::HADMapBin
and others are same message type

**Review by tkimura4 — APPROVED** [review]
LGTM

**Comment by taikitanaka3** [comment]
Note : This package depends on grid map ros

**Comment by taikitanaka3** [comment]
https://github.com/autowarefoundation/autoware.universe/issues/92

**Pull request merged**